src/app/generation.service.ts

```diff
@@ -6,13 +6,31 @@ import { Injectable } from '@angular/core';
   providedIn: 'root'
 })
 export class GenerationService {
-  item$: Observable<any>;
+  items$: Observable<any>;
+  essential$: Observable<any>;
+  filteredItems$: Observable<any>;
+  prop1;
+  prop2;
 
   constructor(private afs: AngularFirestore) {
   }
 
-  getList(type, gender, activities) {
-    return this.item$ = this.afs.collection('pack-items', ref => ref.where('type', '==', type)).valueChanges();
+  getList(weather, type, activities) {
+    this.essential$ = this.afs.collection('list-items', ref => ref.where('activities', '==', 'Essential')).valueChanges();
+
+    if (activities === { baby: true, pet: true, sport: true }) {
+    return this.items$ = this.afs.collection('list-items', ref => ref.where('weather', 'array-contains', weather)
+    .where('type', '==', type)).valueChanges();
+    } else if (activities === { baby: false, pet: false, sport: false }) {
+      this.items$ = this.essential$;
+      return this.items$;
+    } else {
+      this.filteredItems$ = this.afs.collection('list-items', ref => ref.where('weather', 'array-contains', 'average'/*weather*/)
+      .where('type', '==', type)/*.where('activities', '==', activities)*/).valueChanges();
+      this.items$ = this.filteredItems$;
+
+      return this.items$;
+    }
```

Comment:
Please explain to me why do we need these conditions?

Reply:
We don't need it, I'll fix it.